File: prime/src/main/java/org/ostelco/prime/storage/entities/RecordOfPurchase.java
package org.ostelco.prime.storage.entities;

import org.ostelco.prime.storage.AsMappable;

import java.util.HashMap;
import java.util.Map;

public interface RecordOfPurchase extends AsMappable {
    String getMsisdn();

    long   getMillisSinceEpoch();

    String getSku();

    @Override
    default Map<String, Object> asMap() {
        final Map<String, Object> result = new HashMap<>();
        result.put("msisdn", getMsisdn());
        result.put("sku", getSku());
        result.put("timeInMillisSinceEpoch", getMillisSinceEpoch());
        return result;
    }
}

File: prime/src/main/kotlin/org/ostelco/prime/storage/entities/RecordOfPurchase.kt
package org.ostelco.prime.storage.entities

import org.ostelco.prime.storage.AsMappable
import java.util.*

interface RecordOfPurchase : AsMappable {
    val msisdn: String

    val millisSinceEpoch: Long

    val sku: String

    override fun asMap(): Map<String, Any> {
        val result = HashMap<String, Any>()
        result["msisdn"] = msisdn
        result["sku"] = sku
        result["timeInMillisSinceEpoch"] = millisSinceEpoch
        return result
    }
}
